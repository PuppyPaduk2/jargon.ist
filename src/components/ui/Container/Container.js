// @flow

import React, { type Node } from 'react';
import cx from 'classnames';

type Props = {
  children: Node,
  className?: string,
};

const Container = ({ children, className, ...otherProps }: Props) => (
  <div className={cx('container', className)} {...otherProps}>
    {children}
  </div>
);

Container.defaultProps = {
  className: '',
};

export default Container;
