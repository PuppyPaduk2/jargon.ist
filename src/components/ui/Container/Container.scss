@import '../../../styles/abstracts/index';

.c-container {
  margin-left: auto;
  margin-right: auto;
  max-width: px-to-rem(980px);
  padding-left: 1rem;
  padding-right: 1rem;
}
