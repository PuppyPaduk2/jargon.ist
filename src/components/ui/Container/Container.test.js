// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Container from './Container';

it('renders correctly', () => {
  const wrapper = shallow(
    <Container>
      <div>Example content</div>
    </Container>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('passes classname to wrapper', () => {
  const wrapper = shallow(
    <Container className="example-classname">
      <div>Example content</div>
    </Container>,
  );

  expect(wrapper.find('.example-classname').exists()).toEqual(true);
});
