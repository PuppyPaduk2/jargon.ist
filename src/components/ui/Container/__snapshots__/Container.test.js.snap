// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div>
    Example content
  </div>
</div>
`;
